core/src/main/scala/kafka/message/ByteBufferMessageSet.scala

```diff
@@ -78,7 +78,7 @@ object ByteBufferMessageSet {
   }
 
   /** Deep iterator that decompresses the message sets and adjusts timestamp and offset if needed. */
-  def deepIterator(wrapperMessageAndOffset: MessageAndOffset): Iterator[MessageAndOffset] = {
+  def deepIterator(wrapperMessageAndOffset: MessageAndOffset, ignorePartiallyConverted: Boolean = false): Iterator[MessageAndOffset] = {
```

Comment:
I had something close to that for a while, but I thought it might be too general. Do you think we need to handle the case where a v1 message set has been placed in a v0 wrapper? That would mean the user had to downgrade the message format version.